src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -169,6 +169,12 @@ class NetworkOPsImp final
          * @return JSON object.
          */
         StateCountersJson json() const;
+
+        auto getCounterData() const {
```

Comment:
I think I'd prefer the version with the 

```
struct CounterData {
    decltype(counters_) counters;
    decltype(mode_) mode;
    decltype(start_) start;
};
```

Then just `return { counters_, mode_, start_ };`

I realized that `tuple/pair` is great for generic programming, but if you already know the types and can give them reasonable names, then it's better to just declare new struct. Makes it easier to read.